rotkehlchen/rotkehlchen.py

```diff
@@ -746,27 +746,47 @@ def query_balances(
                 if len(loopring_balances) != 0:
                     balances[str(Location.LOOPRING)] = loopring_balances
 
-        # retrieve uniswap v3 lps
+        # retrieve uniswap v3 & v2 lps
         uniswap = self.chain_manager.get_module('uniswap')
-        uniswap_balances = None
+        uniswap_v3_balances = None
         if uniswap is not None:
             try:
-                uniswap_balances = uniswap.get_v3_balances(
+                uniswap_v3_balances = uniswap.get_v3_balances(
                     addresses=self.chain_manager.queried_addresses_for_module('uniswap'),
                 )
             except RemoteError as e:
                 log.error(
                     f'At balance snapshot Uniswap V3 balances query failed due to {str(e)}. Error '
                     f'is ignored and balance snapshot will still be saved.',
                 )
+            try:
+                uniswap_v2_balances = uniswap.get_balances(
+                    addresses=self.chain_manager.queried_addresses_for_module('uniswap'),
+                )
+            except RemoteError as e:
+                log.error(
+                    f'At balance snapshot Uniswap V2 balances query failed due to {str(e)}. Error '
+                    f'is ignored and balance snapshot will still be saved.',
+                )
+            else:
+                if str(Location.BLOCKCHAIN) not in balances:
+                    balances[str(Location.BLOCKCHAIN)] = {}
+
+                for uniswap_v2_lps in uniswap_v2_balances.values():
```

Comment:
This won't work for what we want it to work as.

You are getting the pool's underlying assets only and adding them to the balances and the snapshot. So there is no way for the frontend to show the actual LP tokens and their underlying assets.

Or am I missing something?

Reply:
The way it works for V3 is that there's an extra query the frontend makes to `/uniswap/v3/balances` which provides that(which I replicated in this case)

Now that you mention it, that extra query is unnecessary as it leads to duplication of queries. What do you think about modifying the dictionary returned here to include **liquidity pools**
https://github.com/rotki/rotki/blob/d4408de35e2e913bb2ab871c9e55cd5676dd65d0/rotkehlchen/rotkehlchen.py#L833-L839

Reply:
But wouldn't this make the balance query even slower/longer? Or is this data already queried there?

Reply:
So as discussed we will keep it so the frontend needs to do the extra query. But the backend logic needs to adjust since the graph is getting deprecated